### PR TITLE
print User logger traces in log file

### PR DESCRIPTION
**Pull request by Sefra35S**
File logger to capture the console outputs emitted during a Bender decomposition. (from User Log)

related #178

### CODE REVIEW

File: src/cpp/exe/benders_mpi/main.cpp [hunk 1]
```diff
@@ -36,6 +36,19 @@ int main(int argc, char** argv)
...
+    std::string loggerFileName = options.OUTPUTROOT + PATH_SEPARATOR + "reportbendersmpi";
+    if (world.rank() == 0)
+    {
+        loggerFileName += "-" + std::to_string(world.rank());
+    }
+    Logger loggerFile = std::make_shared<xpansion::logger::UserFile>(loggerFileName);
+    masterLogger->addLogger(loggerUser);
+    masterLogger->addLogger(loggerFile);
+
+    Logger logger = masterLogger;
```
Comment (jmkerloch) [inline src/cpp/exe/benders_mpi/main.cpp:50]: This copy should be discussed later with @sgatto.
This doesn't block the merge so we can deliver 0.4.0 version

File: src/cpp/logger/UserFile.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+#include <list>
+
+#include "logger/UserFile.h"
+
+namespace xpansion {
+    namespace logger {
+
+        UserFile::UserFile(const std::string& filename)
+        {
+            _file.open(filename);
```
Comment (jmkerloch) [inline src/cpp/logger/UserFile.cpp:17]: There will be an exception raised if the file can't be open. Maybe we should add a try catch

File: src/cpp/logger/include/logger/UserFile.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        void log_at_iteration_end(const LogData& d) override;
+
+        void log_at_ending(const LogData& d) override;
+
+        void log_total_duration(double durationInSeconds) override;
+
+    private:
+
+		std::ofstream 		    _file;
+        std::unique_ptr<User>   _userLog;
```
Comment (jmkerloch) [inline src/cpp/logger/include/logger/UserFile.h:43]: Indentation

File: tests/cpp/logger/logger_test.cpp [hunk 4]
```diff
@@ -9,9 +9,73 @@
...
+        stringStreamFromFile << fileStream.rdbuf();
+        fileStream.close();
+    }
+
+    ASSERT_TRUE(stringStreamFromFile.str().empty());
+}
+
+TEST_F(ConsoleLoggerTest, FileHasBeenWritten) {
+    UserFile consoleLog(_fileName);
+    const std::string& displayMessage = "Test d'�criture";
```
Comment (jmkerloch) [inline tests/cpp/logger/logger_test.cpp:62]: Maybe use english test message.

Comment (sgatto) [reply tests/cpp/logger/logger_test.cpp:62]: I agree. No characters avec accents :)

**Review by sgatto — APPROVED** [review]

**Pull request merged**